notebooks/MetPy_Advanced/Isentropic Analysis.ipynb

```diff
@@ -239,27 +240,39 @@
    },
    "outputs": [],
    "source": [
-    "lat = dat.metpy.y\n",
-    "lon = dat.metpy.x\n",
-    "press = ds.metpy.parse_cf('isobaric')\n",
-    "temperature = ds.metpy.parse_cf('Temperature_isobaric')[0]\n",
+    "lat = temperature.metpy.y\n",
+    "lon = temperature.metpy.x\n",
+    "temperature = temperature[0]\n",
     "\n",
     "# Need to adjust units on humidity because '%' causes problems\n",
     "ds['Relative_humidity_isobaric'].attrs['units'] = 'percent'\n",
     "\n",
-    "rh = ds.metpy.parse_cf('Relative_humidity_isobaric')[0]\n",
-    "height = ds.metpy.parse_cf('Geopotential_height_isobaric')[0]\n",
-    "u = ds.metpy.parse_cf('u-component_of_wind_isobaric')[0]\n",
-    "v = ds.metpy.parse_cf('v-component_of_wind_isobaric')[0]\n",
+    "rh = ds['Relative_humidity_isobaric'][0]\n",
+    "height = ds['Geopotential_height_isobaric'][0]\n",
+    "u = ds['u-component_of_wind_isobaric'][0]\n",
+    "v = ds['v-component_of_wind_isobaric'][0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "is_executing": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Can have different vertical levels for wind and thermodynamic variables\n",
+    "# Find and select the common levels\n",
+    "press = temperature.metpy.vertical\n",
+    "common_levels = np.intersect1d(press, u.metpy.vertical)\n",
+    "temperature = temperature.metpy.sel(vertical=common_levels)\n",
+    "u = u.metpy.sel(vertical=common_levels)\n",
+    "v = v.metpy.sel(vertical=common_levels)\n",
     "\n",
-    "# Due to a different number of vertical levels find where they are common\n",
-    "lev_hght = temperature.coords['isobaric4'] * units.Pa\n",
-    "lev_uwnd = u.coords['isobaric'] * units.Pa\n",
-    "_, _, common_ind = np.intersect1d(lev_uwnd, lev_hght, return_indices=True)\n",
-    "temperature = temperature[common_ind, :, :]\n",
-    "# Reassign units to variables\n",
-    "temperature.attrs['units'] = 'kelvin'\n",
-    "lev_uwnd.attrs['units'] = 'Pa'"
+    "# Get common pressure levels as a data array\n",
+    "press = press.metpy.sel(vertical=common_levels)"
```

Comment:
Should we be going back and reworking other examples that use this common levels hack with this method that you have here?

Reply:
Maybe? Probably on a case-by-base basis. Anywhere we're doing something more complicated to handle differing vertical levels.